devel/apache-arrow/Portfile

```diff
@@ -45,13 +47,41 @@ depends_build-append \
 
 depends_lib-append \
                     port:abseil \
-                    port:llvm-${llvm_version}
+                    path:lib/libssl.dylib:openssl \
+                    port:boost \
+                    port:brotli \
+                    port:bzip2 \
+                    port:c-ares \
+                    port:gmake \
+                    port:grpc \
+                    port:google-glog \
+                    port:jemalloc \
+                    port:libutf8proc \
+                    port:libunwind \
+                    port:llvm-${llvm_version} \
+                    port:lz4 \
+                    port:ninja \
+                    port:protobuf3-cpp \
+                    port:rapidjson \
+                    port:re2 \
+                    port:snappy \
+                    port:thrift \
+                    port:wget \
+                    port:zlib \
+                    port:zstd
 
 if { ${os.platform} eq "darwin" && ${os.major} < 17 } {
     known_fail      yes
     pre-fetch {
         ui_error "${name} @${version} requires macOS 10.13 or later."
-        return -code error "incompatible macOS version"
+        return -code error "incompatible OS X version"
+    }
+}
+
+pre-configure {
+    xinstall -d ${workpath}/bin
+    foreach f [glob -nocomplain ${prefix}/bin/ll*-${llvm_version}] {
+        ln -s ${f} ${workpath}/bin/[strsed [file tail ${f}] "s|-${llvm_version}$||"]
```

Comment:
I dont see why this complication is necessary, when you could just directly add the path to the LLVM libexec install area to PATH, as was done in https://github.com/macports/macports-ports/commit/32307f47e04ab9abb61d3d3fa610bfd3440d0a5b#diff-48798ca325280ad7ee3005fd6db459c1b47015fc2397f5809502e4bf38066501 ?

Reply:
> The diffs here are extensive and it is difficult to see what conflict exactly you are fixing ?

A refactor is necessary to address this issue:
```bash
$ python3.8 -c 'import pyarrow.lib as _lib'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/__init__.py", line 63, in <module>
    import pyarrow.lib as _lib
ImportError: dlopen(/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pyarrow/lib.cpython-38-darwin.so, 2): Symbol not found: __Py_FatalErrorFunc
  Referenced from: /opt/local/lib/libarrow_python.300.dylib
  Expected in: flat namespace
 in /opt/local/lib/libarrow_python.300.dylib
```

The `py*-pyarrow` ports must build separately from the `apache-arrow` port.

Reply:
Thanks for doing this the right way. I started working on this refactor before your improvements were added and obviously missed these. The latest push is up-to-date.